sklearn/metrics/_plot/cap_curve.py

```diff
@@ -0,0 +1,274 @@
+        **kwargs,
+    ):
+        """Create a CAP Curve display from an estimator.
+
+        Parameters
+        ----------
+        estimator : estimator instance
+            Fitted classifier or a fitted :class:`~sklearn.pipeline.Pipeline`
+            in which the last estimator is a classifier.
+
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Input values.
+
+        y : array-like of shape (n_samples,)
+            Target values.
+
+        sample_weight : array-like of shape (n_samples,), default=None
+            Sample weights.
+
+        response_method : {'predict_proba', 'decision_function', 'auto'} \
+                default='auto'
+            Specifies whether to use :term:`predict_proba` or
+            :term:`decision_function` as the target response. If set to 'auto',
+            :term:`predict_proba` is tried first and if it does not exist
+            :term:`decision_function` is tried next.
+
+        pos_label : int, float, bool or str, default=None
+            The class considered as the positive class when computing metrics.
+            By default, `estimators.classes_[1]` is considered as the positive class.
+
+        name : str, default=None
+            Name of CAP Curve for labeling. If `None`, use the name of the estimator.
+
+        ax : matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is created.
+
+        **kwargs : dict
+            Keyword arguments to be passed to matplotlib's `plot`.
+
+        Returns
+        -------
+        display : :class:`~sklearn.metrics.CumulativeAccuracyDisplay`
+            The CAP Curve display.
+        """
+        
+        # validate and prepare the prediction scores
+        if response_method == "auto":
+            if hasattr(estimator, "predict_proba"):
+                response_method = "predict_proba"
+            elif hasattr(estimator, "decision_function"):
+                response_method = "decision_function"
+            else:
+                raise ValueError("Estimator does not have a predict_proba or decision_function method.")
+
+        if response_method == "predict_proba":
+            probabilities = estimator.predict_proba(X)
+            
+            # assuming positive class is the second column
+            if pos_label is None:
+                pos_label = 1
+            class_index = np.where(estimator.classes_ == pos_label)[0][0]
+            y_pred = probabilities[:, class_index]
+        elif response_method == "decision_function":
+            y_pred = estimator.decision_function(X)
+
+        if name is None:
+            name = estimator.__class__.__name__
+
+        return cls.from_predictions(
+            y_true=y,
+            y_pred=y_pred,
+            sample_weight=sample_weight,
+            name=name,
+            ax=ax,
+            pos_label=pos_label,
+            **kwargs,
+        )
+
+
+
+import matplotlib.pyplot as plt
```

Comment:
On a first pass, could you add this small test within a separate test file.

Reply:
We indeed need a test (look at how other `*Display` classes are tested in scikit-learn with the following bash command:

```
git grep Display | grep test_
```

or use the search files feature of your IDE.

But it would also be nice to update the last section of following example from the documentation so as to illustrate how to use this class on a concrete case and see the result:

https://scikit-learn.org/stable/auto_examples/linear_model/plot_poisson_regression_non_normal_loss.html

The source code for that example is here:

https://github.com/scikit-learn/scikit-learn/blob/main/examples/linear_model/plot_poisson_regression_non_normal_loss.py

Maybe you could add a `swap_axis` option to allow the user to choose between a CAP/Lift curve or a Lorenz curve.